### PR TITLE
Issue #1273 cleanup example

### LINKED ISSUE #1273: Add example showcasing cleanup utilities
Cleanup utilities were added for #1164.

Unfortunately, I didn't find time yet to add an example for this new functionality. It is good to have some examples, also for the Delft Software Days.

### PR DESCRIPTION
**Pull request by JoerivanEngelen**
Fixes #1273

# Description
Add example of cleanup utilities to the user guide. I think it shows a common use-case for cleaning up data.

Technically the most challenging issue was to work around the fact that sphinx_gallery doesn't have an option to select expected errors in specific code blocks. There is [this PR](https://github.com/sphinx-gallery/sphinx-gallery/pull/930) to introduce this, but unfortunately for us this has not been finished yet. 

I resorted to trapping the error and printing it with a context manager, that seemed like the cleanest solution to me. I added this small context to the public API, so that users can also read in the online documentation what the context manager does.

# Checklist

- [x] Links to correct issue
- [x] Update changelog, if changes affect users
- [x] PR title starts with ``Issue #nr``, e.g. ``Issue #737``
- [x] Unit tests were added
- [x] **If feature added**: Added/extended example

### CODE REVIEW

File: examples/user-guide/10-cleanup.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
+"""
+Data cleanup
+============
+
+More often than not, data contained in databases is not entirely consistent,
+causing errors. It therefore is useful to have some utilities at hand to clean
+up data. We included some convenience methods to help cleaning up inconsistent
```
Comment (Manangka) [inline examples/user-guide/10-cleanup.py:7]: cleaning up => clean up

File: examples/user-guide/10-cleanup.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+
+gwf_simulation = imod.data.hondsrug_simulation(tmpdir / "hondsrug_saved")
+
+
+# %%
+#
+# Update existing model with new data
+# -----------------------------------
+#
+# Your dear colleague has brought you some new data for the river package, which
```
Comment (Manangka) [inline examples/user-guide/10-cleanup.py:36]: I like this :D

File: examples/user-guide/10-cleanup.py [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+#
+# iMOD Python has a set of utilities to help you with cleaning erronous data.
+# Note that these cleanup utilities can only fix a limited set of common
+# mistakes, in ways you might not need/like. It therefore is always wise to
+# verify if you think the fixes are correct in this case.
+#
+# Let's clean up the problematic river data using the
+# :meth:`imod.mf6.River.cleanup` method. Note that this method changes the data
+# inplace. This means that the package's will be updated, without returning a
+# new copy of the package. This means that do comparisons of the
```
Comment (Manangka) [inline examples/user-guide/10-cleanup.py:144]: "that do" = > "that to do"

File: imod/data/sample_data.py [hunk 4]
```diff
@@ -248,3 +248,30 @@ def hondsrug_planar_river() -> xr.Dataset:
...
+    """
+    River data with some mistakes introduced to showcase cleanup
+    utilities.
+    """
+    gwf_model = hondsrug_simulation(path)["GWF"]
+    dis_ds = gwf_model["dis"].dataset
+    riv_ds_old = gwf_model["riv"].dataset
+    # Existing RIV package has only layer coord with [1, 3, 5, 6]. This causes
+    # problems with some methods, at least with cleanup. Therefore align data
+    # here for the cleanup example. Not sure if we want to support river
```
Comment (Manangka) [inline imod/data/sample_data.py:263]: "Not sure if we want to support river packages with limited layer coords"

I'm not sure if we should put the above in the comment.

Instead you could also write:
"River packages with limited layer coords are currently not supported."

**Comment by Manangka** [comment]
I really like this way of writing examples!!

File: imod/data/sample_data.py (hunk 4 above)
Comment (JoerivanEngelen) [reply imod/data/sample_data.py:263]: I agree, I changed it to "River packages with limited layer coords are currently not fully supported.", as you can apparently write a functional model with the river package in only limited layers, but cannot call cleanup on it. I wasn't sure if we want to support that to begin with.

**Pull request merged**